swagger_spec_compatibility/walkers/additional_properties.py

```diff
@@ -39,6 +40,45 @@ def fix_parameter_path(self, path, original_path):
         )
 
 
+CycleSentinel = collections.namedtuple('CycleSentinel', ('path',))
+
+
+def _cycle_safe_dict(dict_with_cycles):
+    # type: (typing.Dict[typing.Any, typing.Any]) -> typing.Dict[typing.Any, typing.Any]
+    """Return a new dict with cycles replaced by sentinels.
+
+    This can be used to do a deep equality comparison against dictionaries
+    which can contain cycles (e.g. expanded Swagger specs).
+    """
+    seen = {}  # type: typing.Dict[int, typing.Tuple[typing.Any, ...]]
```

Comment:
Please address typing

Reply:
Can you elaborate?

Reply:
Sorry wrong line.

The comment was meant for the typing of `process`. Going to open the comment there